src/hivpy/hiv_status.py

```diff
@@ -250,6 +267,172 @@ def stp_HIV_transmission(self, person):
+            risk_to_ltp = rng.normal(
+                self.transmission_rate_means[vl_group],
+                self.transmission_rate_sigmas[vl_group],
+                size=size
+            )
+            if (sex == SexType.Male):
+                if (age_group == 0):
+                    risk_to_ltp *= self.young_women_transmission_factor
+                else:
+                    risk_to_ltp *= self.women_transmission_factor
+                if sti:
+                    risk_to_ltp *= self.sti_transmission_factor
+
+            return rng.uniform(0, 1, size=size) < risk_to_ltp
+
+        people_with_monogamous_ltp_and_hiv = population.get_sub_pop([(col.LONG_TERM_PARTNER, op.eq, True),
+                                                                    (col.LTP_MONOGAMOUS, op.eq, True),
+                                                                    (col.HIV_STATUS, op.eq, True)])
+        partner_infected = population.transform_group([col.SEX, col.AGE_GROUP, col.VIRAL_LOAD_GROUP, col.STI],
+                                                      calculate_infected_ltp_monogamous,
+                                                      use_size=True,
+                                                      sub_pop=people_with_monogamous_ltp_and_hiv)
+        population.set_present_variable(col.LTP_STATUS, partner_infected, people_with_monogamous_ltp_and_hiv)
+
+    def prob_of_infection_from_infected_ltp(self, population: Population):
+        """
+        Sets the probability of infecttion (RISK_LTP)for the population set
+          for which infection occurs from an infected long term partner.
+        """
+        def calculate_risk_of_infection(viral_suppression, ltp_status, size):
+            subgroup = population.get_sub_pop([(col.VIRAL_SUPPRESSION, op.eq, viral_suppression),
+                                               (col.LTP_STATUS, op.eq, ltp_status)])
+            if viral_suppression:
+                risk = rng.normal(self.tr_rate_undetectable_vl, 0.000025, size)
+                vlg = 0
+            else:
+                if ltp_status:
+                    risk = rng.normal((0.05*self.transmission_factor), 0.0125, size)
+                    vlg = 3
+                else:
+                    risk = rng.normal(self.tr_rate_primary, 0.075, size)
+                    vlg = 5
+                    population.set_present_variable(col.LTP_IN_PRIMARY, True, subgroup)
+
+            population.set_present_variable(col.RISK_LTP, risk, subgroup)
+            population.set_present_variable(col.VIRAL_LOAD_GROUP, vlg, subgroup)
+            population.set_present_variable(col.RESISTANCE_MUTATIONS, self.resistance_mutations_prop_vlg[vlg],
+                                            subgroup)
+
+        people_with_ltp = population.get_sub_pop([(col.LONG_TERM_PARTNER, op.eq, True)])
+        population.transform_group([col.VIRAL_SUPPRESSION, col.LTP_STATUS],
```

Comment:
Instead of `col.LTP_STATUS` we might want to use a LTP_INFECTION_DATE or something similar to track whether someone is in primary infection. 

In general we don't want to transform_group by dates so we can have bool column LTP_PRIMARY_INFECTION which before transform_group is set by LTP_PRIMARY_INFECTION = LTP_INFECTION_DATE < (date - length of primary)

Reply:
@mmcleod89 thanks, following your suggestions I transformed this to `ltp_primary_infection = ltp_infection_date > (population.date - timedelta(days=90))` assuming that 90 days is the timestep length in agreement with current timestep of ca.3 months in SAS.